plugins/inputs/postgresql/service.go

```diff
@@ -148,30 +152,32 @@ func (p *Service) Stop() {
 	p.DB.Close()
 }
 
-var kvMatcher, _ = regexp.Compile(`(password|sslcert|sslkey|sslmode|sslrootcert)=\S+ ?`)
+var sanitizer = regexp.MustCompile(`(\s|^)((?:password|sslcert|sslkey|sslmode|sslrootcert)\s?=\s?(?:(?:'(?:[^'\\]|\\.)*')|(?:\S+)))`)
 
 // SanitizedAddress utility function to strip sensitive information from the connection string.
-func (p *Service) SanitizedAddress() (sanitizedAddress string, err error) {
+func (p *Service) SanitizedAddress() (string, error) {
 	if p.OutputAddress != "" {
 		return p.OutputAddress, nil
 	}
 
-	addr, err := p.Address.Get()
+	// Get the address
+	addrSecret, err := p.Address.Get()
 	if err != nil {
-		return sanitizedAddress, fmt.Errorf("getting address for sanitization failed: %w", err)
+		return "", fmt.Errorf("getting address for sanitization failed: %w", err)
 	}
-	defer addr.Destroy()
+	addr := addrSecret.String()
+	addrSecret.Destroy()
 
-	var canonicalizedAddress string
-	if strings.HasPrefix(addr.TemporaryString(), "postgres://") || strings.HasPrefix(addr.TemporaryString(), "postgresql://") {
-		if canonicalizedAddress, err = parseURL(addr.String()); err != nil {
-			return sanitizedAddress, err
+	// Make sure we convert URI-formatted strings into key-values
+	if strings.HasPrefix(addr, "postgres://") || strings.HasPrefix(addr, "postgresql://") {
+		if addr, err = toKeyValue(addr); err != nil {
+			return "", err
 		}
-	} else {
-		canonicalizedAddress = addr.String()
 	}
 
-	return kvMatcher.ReplaceAllString(canonicalizedAddress, ""), nil
+	// Sanitize the string using a regular expression
+	sanitized := sanitizer.ReplaceAllString(addr, "")
```

Comment:
The problem is that we use-up a lot (i.e. three) of locked pages when "getting" secrets, that's why we want to hold them as "short-as-possible". If you are really concerned about holding those, I would switch to "TemporaryString()" but this has the risk of Telegraf using up the locked pages and in turn running into a panic...

Reply:
@srebhan - I see in the updated commits that you are using `TemporaryString()`. This is of course fine with me from a security standpoint, but I was prepared to acquiesce to not using it. I don't recall all the details from before about when we'd run out of locked pages, but I was hoping that in this case it would be a quick open and close and not a leak of locked pages. This is a database plugin and I know that they tend to hold on to things.... I'll let you decide how to handle it now that I've rambled long enough, heh. :)